### DISCUSSION TITLE
Use with React

### DISCUSSION THREAD

**Discussion by PASMarcus** [Eptagone/Vite.AspNetCore#77] [Q&A]
I'm working on quite a big ASP.NET Core MVC project with lots of controllers and razor views and want to begin implement React with Vite slowly in just some of those views/pages. I'm very new to React and Vite and have looked at the example project and I'm not sure how I would go about implementing React using the code snippets in it. Is there an example project where React is implemented too?

Also, it seems like you can't implement React in just some pages and still use the normal razor views for others since Vite only compiles one main.ts file for the whole project. How could you implement a few different React components selectively in just some pages? 

Thanks in advance.

**Comment by Eptagone** [comment]
Hi, of course you can use React with only a few pages. You can use [dynamic imports](https://vite.dev/guide/features#dynamic-import) to load the react app/components in specific elements only if they exist. An also, you can have [multiple entry points](https://vite.dev/config/dep-optimization-options.html#optimizedeps-esbuildoptions) in your Vite configuration.
